kubernetes/proposals/split-k8s-listener.md

```diff
@@ -0,0 +1,19 @@
+# This is a design proposal for changes in k8s-listener.
+
+Roadmap outlined by Juergen suggests we split k8s-listener.py into 2 standalone components.
+First component is to run on Kubernetess master node and dispatch orders to second component,
+running on every kube-node.
+
+* Proposed change is to replace `policy_update` call found in `process` function with
+`dispatch_orders` call that will retrieve ip addresses of all hosts from romana topology
+service, then loop over this addresses and send policy object with romana metadata to
+each host via  HTTP POST.
+* Introduce optional commadline parameter that would trigger `ListenAndServe` function in `main`
+instead of kube-api listener. The only HTTP handler for `ListenAndServe` will accept POST requests,
+parse out `policy_definition` from request and call `policy_update`.
+
+# Open questions
+* I haven't got final response about starting this in Golang right away. Feedback was positive 
+during last meeting but last email from Juergen emphasizes using python to save time. Need final decision.
```

Comment:
I would prefer to just quickly do the split, so that we can have two components, which can then be worked on separately. For example Stas and Greg can work on the Go-port of each half separately. I would like to get into the state where we have a single listener, so that the event handling is more reliable (we can keep track of the time-stamps, I added this to the card, had forgotten that point earlier).